### DISCUSSION TITLE
Add a filetype to an existing lazy autocmd

### DISCUSSION THREAD

**Discussion by moonstruckmoth** [LazyVim/LazyVim#264] [Q&A] [answered]
I would like to add a filetype to the "lazyvim_close_with_q" autogroup.
I looked at nvim_get_autocmds, but I'm not sure if this is the way to go.
Any suggestions?

TIA

**Comment by folke** [comment, marked as answer]
Just copy the code in `lua/config/autocmds.lua`

```lua
vim.api.nvim_create_autocmd("FileType", {
  pattern = {
    "whatever",
  },
  callback = function(event)
    vim.bo[event.buf].buflisted = false
    vim.keymap.set("n", "q", "<cmd>close<cr>", { buffer = event.buf, silent = true })
  end,
})
```

**Comment by moonstruckmoth** [comment]
Doing this overrides the existing LazyVim's existing lazyvim_close_with_q autocmds.
Anyway, I created a new autogroup and added my filetypes there.

**Reply by dpetka2001** [reply]
I think because the group has already been created you should do `group = "lazyvim_close_with_q"` instead of `group = augroup("close_with_q")`.

**Reply by dpetka2001** [reply]
Also the group is called `lazyvim_close_with_q`. If you look at `.local/share/nvim/lazy/lazyvim/lua/lazyvim/config/autocmds.lua` you will see that the augroup function creates the group with the prefix `lazyvim_`.

Edit: you already mentioned this so pls ignore this.

**Reply by dpetka2001** [reply]
I just tried it and it works. It doesn't override the group's default autocmds. It just appends to the pattern list.

**Reply by folke** [reply]
Just don't use the group of course indeed

**Reply by dpetka2001** [reply]
You can still use the group just with the name of the already created group instead of using `augroup(name)` which re-creates the group from scratch, isn't that right? I tested it and it actually appended to the correct augroup if you just use `group = "lazyvim_close_with_q"` which has already been created by lazyvim's default autocmds.

**Reply by folke** [reply]
yep, that also works

**Reply by moonstruckmoth** [reply]
The original solution was correct, I didn't read it properly.
I'm using the group name for consistency.
Thanks everyone for responding.
